src/generator/AutoRest.Go/CodeNamerGo.cs

```diff
@@ -0,0 +1,491 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+
+using AutoRest.Core;
+using AutoRest.Core.Utilities;
+using AutoRest.Core.Utilities.Collections;
+using AutoRest.Core.Model;
+using AutoRest.Go.Model;
+
+namespace AutoRest.Go
+{
+    public class CodeNamerGo : CodeNamer
+    {
+        public new static CodeNamerGo Instance
+        {
+            get
+            {
+                return (CodeNamerGo)CodeNamer.Instance;
+            }
+        }
+
+        // Refactor -> generator
+        public static readonly List<string> AutorestImports = new List<string> { "github.com/Azure/go-autorest/autorest" };
```

Comment:
>ic readonly List<string> [](start = 19, length = 24)

If the intent is to have an immutable set here, this should be declared as IEnumerable<string> :

public static IEnumerable<string> AutorestImports = new string[] { "github.com/Azure/go-autorest/autorest" };

or,  as a Property rather than a static field:

public static IEnumerable<string> AutorestImports {get {  yield return "github.com/Azure/go-autorest/autorest" } };

Either way, the way it's done, the List<string> itself is still mutable.